test/runtests.jl

```diff
@@ -91,19 +91,13 @@ prepend!(tests, linalg_tests)
 
 import LinearAlgebra
 cd(@__DIR__) do
-    # `net_on` implies that we have access to the loopback interface which is
-    # necessary for Distributed multi-processing. There are some test
-    # environments that do not allow access to loopback, so we must disable
-    # addprocs when `net_on` is false. Note that there exist build environments,
-    # including Nix, where `net_on` is false but we still have access to the
-    # loopback interface. It would be great to make this check more specific to
-    # identify those situations somehow. See
-    #   * https://github.com/JuliaLang/julia/issues/6722
-    #   * https://github.com/JuliaLang/julia/pull/29384
-    #   * https://github.com/JuliaLang/julia/pull/40348
     n = 1
-    if net_on
-        n = min(Sys.CPU_THREADS, length(tests))
+    # Allow parallel tests with isolated environment. Some isolated build
+    # environments including Nix or GNU Guix have `net_on` set to false but
+    # still have access to the loopback interface.
+    if net_on || haskey(ENV, "JULIA_CPU_THREADS")
+        x = haskey(ENV, "JULIA_CPU_THREADS") ? parse(Int, ENV["JULIA_CPU_THREADS"]) : Sys.CPU_THREADS
```

Comment:
seems odd to go to all this work to compute the value that you already have
```suggestion
        x = Sys.CPU_THREADS
```

Reply:
Maybe I miss something, but I can use a number of threads different of what my machine physically has.  For instance, one of my machine has 64 cores and I only want to use 32.  This 32 is specified by `JULIA_CPU_THREADS` is I read correctly the doc.

Reply:
`JULIA_CPU_THREADS` is already specified as the way to override Sys.CPU_THREADS

Reply:
Ah indeed.  Thanks.  So only  the `|| haskey(ENV, "JULIA_CPU_THREADS")` seems relevant for building in isolated environment where `net_on` is false but `loopback` is available.

Right?

Reply:
It's a hack that came from NixOS (looking at a container-specific environment variable to override the problematic "net_on").  Ideally net_on should be changed to something for the very thing it requires here, which is a loopback interface.

Reply:
@Apteryks  Yes, but it requires to change `getipaddr` which seems a deep change, where test on `JULIA_CPU_THREADS` seems the documented way to proceed., AFAIU.

```
    net_on = true
    try
        ipa = getipaddr()
    catch
        if ci_option_passed
            @error("Networking unavailable, but `--ci` was passed")
            rethrow()
        end
        net_on = false
        @warn "Networking unavailable: Skipping tests [" * join(net_required_for, ", ") * "]"
        filter!(!in(net_required_for), tests)
    end
```

https://github.com/JuliaLang/julia/blob/master/test/choosetests.jl#L166-L177

Reply:
Yes, it's still a good stopgap solution in my opinion.